### ISSUE TITLE
Data persists in database but not in front-end

### ISSUE THREAD

**Issue by andreasantarl** [ga-wdi-boston/capstone-project#205]
I'm trying to create a GET action with user profile.  I can successfully create a user profile (although I think the data is getting scaffolded incorrectly because the JSON appears as 
user ->profiles -> profile ->data, and it should only be user -> profiles), but then when I try to display that data through a GET request and format the request based on the scaffolding, it says that it cannot find the id of the profile.  I'm tried basically every permutation of setting the code from the POST request and creating new routes based upon the profile id, and nothing seems to work.  Here's my code:

Events
```const onCreateMyProfile = (event) => {

  event.preventDefault();
  let data = getFormFields(event.target);
//  console.log(data);
  // data.profiles.user = app.id;
  api.createMyProfile(data)
  .then(ui.createMyProfileSuccess)
  .catch(error => console.error(error))
};

const onShowMyProfile = () => {
  // console.log(app);
  return api.showMyProfile()
    .then(ui.showMyProfileSuccess)
    .catch(error => console.error(error))
};```

API

```
const createMyProfile = (data) => {
  return new Promise((resolve, reject) => {
    $.ajax({
      url: app.host + '/profiles',
      method: "POST",
      headers: {
        Authorization: 'Token token=' + app.user.token,
      },
      data: data,
      success: (response) => {
        resolve(response);
      },
      error: (error) => {
        reject(error);
      },
    });
  });
};

const showMyProfile = () => {
  console.log(app);
  return new Promise((resolve, reject) => {
    $.ajax({
      url: app.host + '/profiles/' + app.profiles.id,
      method: 'GET',
      headers: {
        Authorization: 'Token token=' + app.user.token,
      },
      success: (response) => {
        resolve(response);
      },
      error: (error) => {
        reject(error);
      },
    });
  });
};
```

UI
```const createMyProfileSuccess = (data) => {
    app.user.profiles = data;
    console.log(app);
};

const showMyProfileSuccess = () => {
  $('#view-profile').html("hello");
  $('#view-profile').html(profileTemplate(data.profile));
};

```
```

**Comment by payne-chris-r** [comment]
Have you tried curl requests? What does the JSON look like if you go to localhost:3000/users/2/profile?

**Comment by andreasantarl** [comment]
Still having the same issue where I can't access the data in the front-end if I sign out and then sign back in.

**Comment by andreasantarl** [comment]
Fixed.  User Login Serializer was being used instead of User Serializer.  Needed to run GET request on successful sign-in to update app.user = data.user

**Comment by jrhorn424** [comment]
Just spoke with @andreasantarl about this. In her client, she was storing deeply-nested properties, which led to some confusion on my part about pluralization, nesting order, etc. Imagine `app.user.profile.id` versus `app.user.profiles` versus `app.profile.user.id`. For this, I suggested storing flat pieces of information on AJAX success, such as `app.userProfileId`.

Additionally, the data that we needed on the front-end to display the profile wasn't coming back from the API (the profile id associated with that user). `UserSerializer` is what we were editing, but `UserLoginSerializer` is what our authentication code uses to serialize a response.

Then, we requested the profile data after successful sign-in using a Promise chain.
